src/main/java/alfio/job/executor/AssignTicketToSubscriberJobExecutor.java

```diff
@@ -0,0 +1,149 @@
+import java.time.temporal.ChronoUnit;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static alfio.model.system.ConfigurationKeys.GENERATE_TICKETS_FOR_SUBSCRIPTIONS;
+
+@Component
+public class AssignTicketToSubscriberJobExecutor implements AdminJobExecutor {
+
+    private final AdminReservationRequestManager requestManager;
+    private final ConfigurationManager configurationManager;
+    private final SubscriptionRepository subscriptionRepository;
+    private final EventRepository eventRepository;
+    private final ClockProvider clockProvider;
+    private final TicketCategoryRepository ticketCategoryRepository;
+
+    public AssignTicketToSubscriberJobExecutor(AdminReservationRequestManager requestManager,
+                                               ConfigurationManager configurationManager,
+                                               SubscriptionRepository subscriptionRepository,
+                                               EventRepository eventRepository,
+                                               ClockProvider clockProvider,
+                                               TicketCategoryRepository ticketCategoryRepository) {
+        this.requestManager = requestManager;
+        this.configurationManager = configurationManager;
+        this.subscriptionRepository = subscriptionRepository;
+        this.eventRepository = eventRepository;
+        this.clockProvider = clockProvider;
+        this.ticketCategoryRepository = ticketCategoryRepository;
+    }
+
+    @Override
+    public Set<JobName> getJobNames() {
+        return EnumSet.of(JobName.ASSIGN_TICKETS_TO_SUBSCRIBERS);
+    }
+
+    @Override
+    public String process(AdminJobSchedule schedule) {
+        // 1. Find all subscriptions bought until now. Filters:
+        //     - subscription_descriptor_fk is linked to the current event
+        //     - id does not have any reservations attached to the current event
+        //     - validity_from is <= now()
+        //     - validity_to is null or > now()
+        //     - status = 'ACQUIRED'
+        var subscriptionsByEvent = subscriptionRepository.loadAvailableSubscriptionsByEvent()
+            .stream()
+            .collect(Collectors.groupingBy(AvailableSubscriptionsByEvent::getEventId));
+        if (!subscriptionsByEvent.isEmpty()) {
+            eventRepository.findByIds(subscriptionsByEvent.keySet()).forEach(event -> {
+                // 2. for each event check if the flag is active
+                boolean generationEnabled = configurationManager.getFor(GENERATE_TICKETS_FOR_SUBSCRIPTIONS, event.getConfigurationLevel())
+                    .getValueAsBooleanOrDefault();
+                if (generationEnabled) {
+                    var subscriptions = subscriptionsByEvent.get(event.getId());
+                    var categories = ticketCategoryRepository.findAllTicketCategories(event.getId());
```

Comment:
We need a better logic for selecting the ticket category.

Should be active, not empty, available to the public.